src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -1273,6 +1273,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return If(arguments.HasValue, arguments.Value, Nothing)
         End Function
 
+        Public Function GetTypeArgumentsOfInvocationExpression(invocationExpression As SyntaxNode) As SeparatedSyntaxList(Of SyntaxNode) Implements ISyntaxFactsService.GetTypeArgumentsOfInvocationExpression
+            Dim arguments = TryCast(TryCast(invocationExpression, InvocationExpressionSyntax)?.Expression, GenericNameSyntax)?.TypeArgumentList?.Arguments
+            Return If(arguments.HasValue, arguments.Value, Nothing)
```

Comment:
this is an example of why this is wrong.  say you have "foo.Bar(of X)(y, z, w)".  This code will not work because the Expression is not a GenericNameSyntax, it's a MemberAccessExpressionSyntax.  But you only encoded the full decomposition in the C# side of things.

Instead, the decomposition should move into AbstractAddParameterXXX